### DISCUSSION TITLE
Fix MC-258430

### DISCUSSION THREAD

**Discussion by Pantera07** [PurpurMC/Purpur#1217] [__Completed__] [answered]
https://bugs.mojang.com/browse/MC-258430
https://github.com/PaperMC/Paper/issues/8640

Paper has no plans to fix this issue. So it would be nice to add a patch to Purpur to fix this issue.

**Comment by BillyGalbreath** [comment, marked as answer]
~~Added a fix in build 1865 https://github.com/PurpurMC/Purpur/commit/2e9fa3114aa7b272aa6cafed681c9ab81fa80~~

Ok, applied a proper fix in build 1867.

Turns out that Paper is sending the wrong gametime to the client, which is used to calculate camel's sit/standing pose tick delta, which is used to calculate the player's mounted offset height.

**Reply by Pantera07** [reply]
I tried testing on the 1866 build, but this issue is still not resolved.
My test world's gametime is 510614

**Reply by BillyGalbreath** [reply]
Hmm, so it's not just the LastPostTime. I'll investigate.

**Reply by Pantera07** [reply]
Camel walks fast without walking animation (probably a vanilla issue)
https://youtu.be/XIiH2Fii40Q

**Reply by BillyGalbreath** [reply]
The walking animation would be controlled client side. Nothing we can fix server side for that.

**Reply by BillyGalbreath** [reply]
Ok, applied a proper fix in build 1867.

Turns out that Paper is sending the _wrong_ gametime to the client, which is used to calculate camel's sit/standing pose tick delta, which is used to calculate the player's mounted offset height.

**Reply by BillyGalbreath** [reply]
Also, i think this fixed the weird walking animation.. I havent seen the behavior since the last fix.

**Reply by Pantera07** [reply]
Yeah camel issue solved. Thanks!
